pkg/volume/plugins.go

```diff
@@ -89,6 +89,14 @@ type RecyclableVolumePlugin interface {
 	NewRecycler(spec *Spec) (Recycler, error)
 }
 
+// CreatableVolumePlugin is an extended interface of VolumePlugin and is used to create persistent volumes for the cluster.
+type CreatableVolumePlugin interface {
+	VolumePlugin
+	// NewCreater creates a new volume.Creater which knows how to create this resource
+	// in accordance with the underlying storage provider
+	NewCreater(spec *Spec) (Creater, error)
```

Comment:
I think this is going to need to take information about the pod spec, too.  See #12944